src/components/Collections/TokenDetail/index.tsx

```diff
@@ -148,97 +162,133 @@ function TokenDetail({ contractAddress, tokenId }: TokenDetailProps) {
+        scrollBehavior="inside"
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalCloseButton />
+          <TokenImage
+            src={ipfsUriToGatewayUrl(system.config.network, token.artifactUri)}
+          />
+        </ModalContent>
+      </Modal>
+      <Flex pt={8} px={8}>
+        <MinterButton
+          variant="primaryActionInverted"
+          onClick={e => {
+            e.preventDefault();
+            setLocation('/collections', { replace: true });
+          }}
+        >
+          <Box color="currentcolor">
+            <ChevronLeft size={16} strokeWidth="3" />
+          </Box>
+          <Text ml={2}>Collections</Text>
+        </MinterButton>
+      </Flex>
+      <Flex
+        align="center"
+        flex="1"
+        pb={[4, 16]}
+        px={[4, 16]}
+        mx="auto"
+        width={['90%', '70%']}
+        flexDir="column"
+      >
+        <TokenImage
+          src={ipfsUriToGatewayUrl(system.config.network, token.artifactUri)}
+        />{' '}
+        <Flex align="center" justify="space-evenly" width={['100%']} mt="4">
+          {isOwner ? (
+            <Menu>
+              <MinterMenuButton variant="primary">
+                <MoreHorizontal color="#25282B" />
+              </MinterMenuButton>
+              <MenuList
+                borderColor="brand.lightBlue"
+                borderRadius="2px"
+                p={0}
+                minWidth={[100]}
+              >
+                <MinterMenuItem
+                  w={[100]}
+                  variant="primary"
+                  onClick={disclosure.onOpen}
+                >
+                  Transfer
+                </MinterMenuItem>
+              </MenuList>
+              <TransferTokenModal
+                contractAddress={contractAddress}
+                tokenId={tokenId}
+                disclosure={disclosure}
               />
-            </Box>
-          </AspectRatio>
+            </Menu>
+          ) : null}
+
+          {token.sale ? (
+            isOwner ? (
+              <>
+                <CancelTokenSaleButton
+                  contract={contractAddress}
+                  tokenId={tokenId}
+                />
+              </>
+            ) : (
+              <Flex
+                flexDirection="column"
+                align="stretch"
+                width={['100%', 200]}
+              >
+                <Flex align="center" alignSelf="center">
+                  <Text color="black" fontSize="3xl" mr={1}>
+                    ꜩ
+                  </Text>
+                  <Text color="brand.black" fontSize="xl" fontWeight="700">
+                    {token.sale.price.toFixed(2)}
+                  </Text>
+                </Flex>
+                <BuyTokenButton contract={contractAddress} token={token} />
+              </Flex>
+            )
+          ) : (
+            <>
+              <SellTokenButton contract={contractAddress} tokenId={tokenId} />
+            </>
```

Comment:
Didn't catch this earlier. Sell token button should only appear for the token owner.